### PR TITLE
feat: [POC] add tooltip feature (SDKCF-4715)

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
this is only a POC.

Still TODO:
**PENDED**: handling for `NestedScrollView` and `ListView`
**PENDED**: removing campaigns when tapping device home button
**PENDED**: hadow for close button

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Comment by donnie-jp** [comment]
You can target this to a `feat/tooltip-poc` branch. It shouldn't be merged to master.

**Review by leila-gu — COMMENTED** [review]
LGTM, just question to get better understanding of the tooltip view.

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/coroutine/MessageActionsCoroutine.kt [hunk 1]
```diff
@@ -227,5 +221,19 @@ internal class MessageActionsCoroutine(
...
+        /**
+         * This method returns which button was clicked which is represented by ImpressionType object.
+         */
+        internal fun getOnClickBehaviorType(viewResourceId: Int): ImpressionType {
+            return when (viewResourceId) {
+                R.id.message_close_button -> ImpressionType.EXIT
+                R.id.message_single_button, R.id.message_button_left -> ImpressionType.ACTION_ONE
+                R.id.message_button_right -> ImpressionType.ACTION_TWO
+                R.id.slide_up, R.id.message_tooltip_image_view, R.id.message_tip -> ImpressionType.CLICK_CONTENT
+                BACK_BUTTON -> ImpressionType.EXIT
```
[the remark this answers is not in the document]
Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/coroutine/MessageActionsCoroutine.kt:234]: merged

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt [hunk 2]
```diff
@@ -33,23 +43,128 @@ internal interface DisplayManager {
...
-            if (inAppMessageBaseView != null) {
-                // Removing just the InApp Message from the view hierarchy.
-                val parent = inAppMessageBaseView.parent as ViewGroup
+            return if (id != null) {
+                // id is not null if tooltip
+                activity.findViewById<ViewGroup>(R.id.in_app_message_tooltip_view)?.parent?.let {
+                    for (i in 0..(it as ViewGroup).childCount) {
+                        val child = it.getChildAt(i)
+                        if (child?.id == R.id.in_app_message_tooltip_view && child.tag == id) {
+                            scheduleRemoval(delay, child as ViewGroup, id, activity)
```
[the remark this answers is not in the document]
Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/DisplayManager.kt:60]: It is possible to have multiple tooltip campaigns displayed. This is to check which tooltip campaign should be displayed using the `tag` value.
Updated logic.

**Review by leila-gu — APPROVED** [review]

**Pull request merged**